### PR TITLE
Update example of building dll on Windows

### PR DESCRIPTION
**Pull request by meteorcloudy**

### CODE REVIEW

File: examples/windows/dll/hello-library.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#ifndef HELLO_LIBRARY_H
+#define HELLO_LIBRARY_H
+
+#if COMPILING_DLL
```
Comment (laszlocsomor) [inline examples/windows/dll/hello-library.h]: nit: `#ifdef`

File: examples/windows/dll/BUILD [hunk 2]
```diff
@@ -4,18 +4,69 @@ filegroup(
...
     linkshared = 1,
 )
 
+
+# **Explicitly link to hellolib.dll**
+
+# Declare hellolib.dll as data dependency and load it explicitly in code.
 cc_binary(
-    name = "hello",
+    name = "hello-explicitly",
```
Comment (laszlocsomor) [inline examples/windows/dll/BUILD]: I'm confused by this name; the word "explicitly" doesn't explain enough for me. Could you choose a different name, which is more concrete? How about e.g. "hello_load-dll-at-runtime" or something similar?

File: examples/windows/dll/BUILD [hunk 3]
```diff
@@ -4,18 +4,69 @@ filegroup(
...
+    name = "hellolib_dll",
+    deps = [
+        ":hellolib_dll_import",
+        ":hello-library-header",
+    ],
+)
+
+# Just depend on :hellolib_dll
+cc_binary(
+    name = "hello-implicitly",
```
Comment (laszlocsomor) [inline examples/windows/dll/BUILD]: Here again, I'm confused by the name. How about e.g. "hello_link-to-dll-via-lib"?

**Comment by laszlocsomor** [comment]
LGTM, because I only had comments about naming and a "nit" comment.

File: examples/windows/dll/hello-library.h (hunk 1 above)
Comment (meteorcloudy) [reply examples/windows/dll/hello-library.h]: Thanks! Addressed in the internal change.

File: examples/windows/dll/BUILD (hunk 2 above)
Comment (meteorcloudy) [reply examples/windows/dll/BUILD]: Same here, addressed in the internal change.

File: examples/windows/dll/BUILD (hunk 3 above)
Comment (meteorcloudy) [reply examples/windows/dll/BUILD]: Same here, addressed in the internal change.

**Pull request closed without merging**